File: solutions/java/src/main/java/playground/algorithm/ContainerWithMostWater.java
package playground.algorithm;

public class ContainerWithMostWater {

    public int maxArea(int[] height) {
        int max = 0;
        for (int l = 0, r = height.length - 1; l < r; ) {
            max = Math.max(max, (r - l) * Math.min(height[l], height[r]));
            if (height[r] > height[l]) {
                l++;
            } else {
                r--;
            }
        }
        return max;
    }
}

File: solutions/go/src/playground/algorithm/container_with_most_water.go
package algorithm

import "math"

func maxArea(height []int) int {
	max := 0
	for l, r := 0, len(height)-1; l < r; {
		max = int(math.Max(float64(max), float64(r-l)*math.Min(float64(height[l]), float64(height[r]))))
		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return max
}
